### PR TITLE
remove requirement that SSL be used on default http port

### PR DESCRIPTION
**Pull request by emartin-merrill-r7**
## Proposed Changes
This PR removes the mismatch between the default behavior of port 5985 and the plugin.  The plugin now checks if it's working on port 5985

### Description

Describe the proposed changes:

  - check if port is 5985 and if it is, don't require ssl

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/powershell/komand_powershell/util/util.py [hunk 1]
```diff
@@ -40,7 +40,8 @@ def local(action, powershell_script):
 def ntlm(action, host_ip, powershell_script, username, password, port):
     # Adds needed https and port number to host IP
     action.logger.info("Running a NTLM connection")
-    host_connection = "https://{host_ip}:{port}/wsman".format(host_ip=host_ip, port=port)
+    prefix = 'http' if port == 5985 else 'https'
```
Comment (ablakley-r7) [inline plugins/powershell/komand_powershell/util/util.py:43]:
```suggestion
    prefix = "http" if port == 5985 else "https"
```

File: plugins/powershell/komand_powershell/util/util.py [hunk 2]
```diff
@@ -76,7 +77,8 @@ def ntlm(action, host_ip, powershell_script, username, password, port):
 def kerberos(action, host_ip, kdc, domain, host_name, powershell_script, password, username, port):
     action.logger.info("Running Kerberos connection")
     # Adds needed https and port number to host IP
-    host_connection = "https://{host_ip}:{port}/wsman".format(host_ip=host_ip, port=port)
+    prefix = 'http' if port == 5985 else 'https'
```
Comment (ablakley-r7) [inline plugins/powershell/komand_powershell/util/util.py:80]:
```suggestion
    prefix = "http" if port == 5985 else "https"
```

File: plugins/powershell/komand_powershell/util/util.py [hunk 3]
```diff
@@ -164,7 +166,8 @@ def kerberos(action, host_ip, kdc, domain, host_name, powershell_script, passwor
 def credssp(action, host_ip, powershell_script, username, password, port):
     # Adds needed https and port number to host IP
     action.logger.info("Running a CredSSP connection")
-    host_connection = "https://{host_ip}:{port}/wsman".format(host_ip=host_ip, port=port)
+    prefix = 'http' if port == 5985 else 'https'
```
Comment (ablakley-r7) [inline plugins/powershell/komand_powershell/util/util.py:169]:
```suggestion
    prefix = "http" if port == 5985 else "https"
```

File: plugins/powershell/help.md [hunk 4]
```diff
@@ -17,7 +17,7 @@
 
 Check out the [plugin guide](https://docs.rapid7.com/insightconnect/mass-delete-with-powershell/) for more details on how to configure this plugin.
 
-The connection configuration accepts the following parameters:
+The connection configuration accepts the following parameters.  SSL is enforced for all ports except 5985:
```
Comment (ablakley-r7) [inline plugins/powershell/help.md:20]:
```suggestion
The connection configuration accepts the following parameters. SSL is enforced for all ports except 5985:
```

**Review by ablakley-r7 — CHANGES_REQUESTED** [review]
Just a ' -> " change and everything else looks great.

**Review by ablakley-r7 — APPROVED** [review]

**Review by cmcnally-r7 — APPROVED** [review]

**Review by mberezin-r7 — APPROVED** [review]

**Comment by mberezin-r7** [comment]
This PR targets `master` directly and skips `develop`. Please target this PR to the `develop` branch and make a subsequent release PR from `develop` to `master`.

**Review by igorski-r7 — APPROVED** [review]

**Review by cmcnally-r7 — APPROVED** [review]

**Review by mberezin-r7 — APPROVED** [review]

**Pull request merged**